File: solutions/cpp/sum-of-multiples/1/sum_of_multiples.cpp
#include "sum_of_multiples.h"

namespace sum_of_multiples {
    int to(vector<int> items, int lim) {
        int sum = 0;
        for (int i = 1; i < lim; ++i) {
            for (auto val : items) {
                if (i % val == 0) {
                    sum += i;
                    break;
                }
            }
        }
        return sum;
    }
}  // namespace sum_of_multiples

File: solutions/java/sum-of-multiples/1/src/main/java/SumOfMultiples.java
import java.util.HashSet;
import java.util.Iterator;

class SumOfMultiples {
    private int number;
    private int set[];
    
    SumOfMultiples(int number, int[] set) {
        this.number = number;
        this.set = set;
    }

    int getSum() {
        HashSet<Integer> multiples = new HashSet<Integer>();
        for (int n: this.set)
            this.fillSeqFor(n, multiples);
        
        Iterator<Integer> iter = multiples.iterator();

        int sum = 0;
        while (iter.hasNext())
            sum += iter.next();

        return sum;
    }

    private void fillSeqFor(int n, HashSet<Integer> multiples) {
        for (int i = n; i < this.number; i += n)
            multiples.add(i);
    }

}